src/test/scala/mesosphere/marathon/state/GroupTest.scala

```diff
@@ -376,5 +376,26 @@ class GroupTest extends FunSpec with GivenWhenThen with Matchers {
       Then("the cycle is detected")
       current.hasNonCyclicDependencies should equal(false)
     }
+
+    it("can contain a path which has the same name multiple times in it") {
```

Comment:
In order to avoid regressions and to make sure that your change fixes the issue, you should add a test that fails without your changes.

I added this test (without the other changes in the commit) to the current master and it passes.